modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -145,6 +153,51 @@ func isAllowedAddress(ctx sdk.Context, receiver string, allowedAddrs []string) b
 	return false
 }
 
+// validateMemo returns a nil error indicating if the memo is valid for transfer.
+func validateMemo(ctx sdk.Context, memo string, allowedPacketDataList []string) error {
+	// if the allow list is empty, then the memo must be an empty string
+	if len(allowedPacketDataList) == 0 {
+		if len(strings.TrimSpace(memo)) != 0 {
+			return errorsmod.Wrapf(ErrInvalidAuthorization, "memo not allowed in empty allowedPacketDataList")
+		}
+
+		return nil
+	}
+
+	// if allowedPacketDataList has only 1 element and it equals AllowAllPacketDataKeys
+	// then accept all the packet data keys
+	if len(allowedPacketDataList) == 1 && allowedPacketDataList[0] == AllowAllPacketDataKeys {
+		return nil
+	}
+
+	jsonObject := make(map[string]interface{})
+	err := json.Unmarshal([]byte(memo), &jsonObject)
+	if err != nil {
+		return err
+	}
+
+	if len(jsonObject) > len(allowedPacketDataList) {
+		return errorsmod.Wrapf(ErrInvalidAuthorization, "packet contains more packet data keys than packet allow list has")
+	}
```

Comment:
The issue I see of doing this is that we don't give any information to the user of what keys are the ones that are not allowed? cc @DimitrisJim

Reply:
So you're fine if I drop this code as well?

Reply:
>I mean the printing of the keys was dropped!

Ay, true! I can put that back and drop this for now then.

Reply:
I will tackle this in a follow-up because the keys were shown before using the experimental feature, so I will try getting the keys in a different way.